services/src/test/java/io/scalecube/services/CoarseGrainedServiceImpl.java

```diff
@@ -1,16 +1,27 @@
 package io.scalecube.services;
 
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
 
+import io.scalecube.services.annotations.Inject;
 import io.scalecube.services.annotations.ServiceProxy;
+import io.scalecube.services.routing.RoundRobinServiceRouter;
 
 public class CoarseGrainedServiceImpl implements CoarseGrainedService {
 
-  @ServiceProxy
+  @ServiceProxy(router = RoundRobinServiceRouter.class, timeout = 3, timeUnit = TimeUnit.SECONDS)
   private GreetingService greetingService;
 
+  @Inject
+  private Microservices ms;
+
   @Override
   public CompletableFuture<String> callGreeting(String name) {
-    return this.greetingService.greeting(name);
+    if (ms != null) {
```

Comment:
I don't really understand the use case. You don't use microservices other than checking it is not null - what is the meaning of it? Maybe it is simpler just not inject `greetingService` if microservices not set and then check that `greetingService` not null.